include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl_utils.h

```diff
@@ -381,15 +381,12 @@ struct __local_buffer<sycl::buffer<::std::tuple<_T...>, __dim, _AllocT>>
     using type = sycl::buffer<oneapi::dpl::__internal::tuple<_T...>, __dim, _AllocT>;
 };
 
-template <typename _ExecutionPolicy, typename _T, typename _Container>
-class __buffer_impl;
-
 // impl for sycl::buffer<...>
-template <typename _ExecutionPolicy, typename _T, typename _BValueT, int __dim, typename _AllocT>
-class __buffer_impl<_ExecutionPolicy, _T, sycl::buffer<_BValueT, __dim, _AllocT>>
+template <typename _ExecutionPolicy, typename _Container>
```

Comment:
It seems a general type `_Container` doesn't  make sense here, because it can be only `sycl::buffer`.
So, I would suggest removing `_Container` template parameter and use `sycl::buffer` directly in `__buffer_impl` class.

Reply:
Fixed